build/beforeBuildSave/BeforeBuildSaveTest.groovy

```diff
@@ -1,19 +1,43 @@
 import spock.lang.Specification
-
+import org.jfrog.artifactory.client.model.repository.settings.impl.MavenRepositorySettingsImpl
+import spock.lang.Specification
 import static org.jfrog.artifactory.client.ArtifactoryClient.create
+import groovy.json.JsonSlurper
+import org.jfrog.artifactory.client.ArtifactoryRequest
+import org.jfrog.artifactory.client.impl.ArtifactoryRequestImpl
+import groovy.json.*
 
 class BeforeBuildSaveTest extends Specification {
-    def 'test name'() {
+
+    def 'before build save test'() {
         setup:
+
         def baseurl = 'http://localhost:8088/artifactory'
-        def artifactory = create(baseurl, 'admin', 'password')
+        def artifactory = create(baseurl, 'admin', 'password') D
+        def auth = "Basic ${'admin:password'.bytes.encodeBase64()}"
 
         when:
-        null
+
+        def conn = new URL("${baseurl}/api/build").openConnection()
+        conn.requestMethod = 'PUT'
+        conn.doOutput = true
+        conn.setRequestProperty('Authorization', auth)
+        conn.setRequestProperty('Content-Type', 'application/json')
+        def jsonfile = new File('./src/test/groovy/BeforeBuildSaveTest/build.json')
```

Comment:
@AngelloMaggio the build.json file being used by this test case must be included in this pull request.